### PR TITLE
Introduce `xchain-bitcoin`

### LINKED ISSUE #574: Switch to `@xchainjs/xchain-bitcoin`
to get rid of `asgardex-bitcoin`

Source of `@xchainjs/xchain-bitcoin`: https://github.com/xchainjs/xchainjs-lib/blob/master/packages/xchain-binance

### PR DESCRIPTION
**Pull request by veado**
- [x] Remove `asgardex-bitcoin` in favour of `xchain-bitcoin` 
- [x] Use `REACT_APP_BLOCKCHAIR_API_KEY` from GH secrets while creating binaries

Closes #574

### CODE REVIEW

File: src/renderer/i18n/ru/wallet.ts [hunk 1]
```diff
@@ -38,9 +38,9 @@ const wallet: WalletMessages = {
   'wallet.receive.address.error': 'Нет доступных адресов для получения',
   'wallet.receive.address.errorQR': 'Ошибка при создании QR-кода: {error}',
   'wallet.send.success': 'Transaction succeeded - RU.',
+  'wallet.send.fastest': 'Fasterst - RU',
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/wallet.ts:41]: 'wallet.send.success': 'Тразакция завершена',
'wallet.send.fastest': 'Быстро',

File: src/renderer/i18n/ru/wallet.ts [hunk 2]
```diff
@@ -38,9 +38,9 @@ const wallet: WalletMessages = {
...
   'wallet.send.fast': 'Быстро',
-  'wallet.send.regular': 'Обычно',
-  'wallet.send.slow': 'Медленно',
+  'wallet.send.average': 'Average - RU',
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/wallet.ts:43]: 'wallet.send.average': 'Средне',

File: src/renderer/services/bitcoin/balances.ts [hunk 3]
```diff
@@ -1,32 +1,24 @@
...
-        RD.success({
-          asset: AssetBTC,
-          amount: baseAmount(balance, BTC_DECIMAL),
-          frozenAmount: O.none
-        } as AssetWithBalance)
-      )
-    ),
+    // Remove transformation `Balance` -> `AssetWithBalance` to use `Balance` only
+    // https://github.com/thorchain/asgardex-electron/issues/584
+    map(FP.flow(A.map((balance) => ({ ...balance, frozenAmount: O.fromNullable(balance.frozenAmount) })))),
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/bitcoin/balances.ts:20]: seems `flow` is not needed here?

File: src/renderer/views/wallet/send/SendViewBTC.tsx [hunk 4]
```diff
@@ -59,27 +58,28 @@ export const SendViewBTC: React.FC<Props> = (props): JSX.Element => {
...
   return FP.pipe(
-    sequenceTOption(oBtcAssetWB, oExplorerUrl),
+    sequenceTOption(oClient, oBtcAssetWB),
     O.fold(
       () => <></>,
-      ([btcAssetWB, explorerUrl]) => {
-        const successActionHandler = (txHash: string) => window.apiUrl.openExternal(`${explorerUrl}tx/${txHash}`)
-
+      ([client, btcAssetWB]) => {
+        const successActionHandler = FP.flow(client.getExplorerTxUrl, window.apiUrl.openExternal)
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/views/wallet/send/SendViewBTC.tsx:72]: i believe this is neccessary to type `successActionHandler` as  `flow` here is outside of any `pipe` and final result is almost untyped here

File: src/renderer/services/wallet/types.ts [hunk 5]
```diff
@@ -102,13 +102,16 @@ export enum ErrorId {
...
+// TODO(@Veado) Move type to clients/type
+
 export type ApiError = {
   errorId: ErrorId
   msg: string
 }
 
 export type NonEmptyApiErrors = NonEmptyArray<ApiError>
 
+// TODO (@thatThorchainGuy) Remove TxRD/TxLD with https://github.com/thorchain/asgardex-electron/issues/573
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/wallet/types.ts:114]: hm. dont see a reason here to remove this values in terms of #573. could you clarify it ?

File: src/renderer/services/wallet/balances.ts [hunk 6]
```diff
@@ -133,6 +128,10 @@ export const assetsWBState$: Observable<AssetsWithBalanceState> = Rx.combineLate
       // filter results out
       // Transformation: RD<Error, AssetsWithBalance>`-> `AssetsWithBalance)[]`
       A.filterMap(RD.toOption),
+      (v) => {
+        console.log('assetsWBState$', v)
+        return v
+      },
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/wallet/balances.ts:134]: oops :)

**Review by thatStrangeGuyThorchain — CHANGES_REQUESTED** [review]
LGTM!
but could you pls fix compile issues

File: src/renderer/views/wallet/send/SendViewBTC.tsx (hunk 4 above)
Comment (veado) [reply src/renderer/views/wallet/send/SendViewBTC.tsx:72]: A type can be still inferred:

hover_type

But I can add a type of course ...

File: src/renderer/services/wallet/types.ts (hunk 5 above)
Comment (veado) [reply src/renderer/services/wallet/types.ts:114]: Good find! I can't remember why I have put this comment there, it does not make sense. _I will remove it ..._

File: src/renderer/services/chain/transaction.ts [hunk 7]
```diff
@@ -47,4 +50,24 @@ const sendStakeTx = ({ chain, asset, poolAddress, amount, memo }: SendStakeTxPar
...
+    O.fold(
+      () => Rx.of(O.none),
+      ({ chain }) =>
+        FP.pipe(
+          clientByChain$(chain),
+          RxOp.map(
+            O.map((client) =>
+              // we leave `txID` parameter empty to get just the url,
+              // but w/o the hash- the hash will be add by the view
+              client.getExplorerTxUrl('')
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/services/chain/transaction.ts]: Reverted with 574d178a5093634891daf7513192fcc6934df816 and 574d178a5093634891daf7513192fcc6934df816. `xchain-bitcoin` has been fixed with https://github.com/xchainjs/xchainjs-lib/pull/97

**Pull request merged**